### DISCUSSION TITLE
Current Source Code Compiling Problem

### DISCUSSION THREAD

**Discussion by I-wish-2** [mriscoc/Ender3V2S1#243] [General]
Greetings,  
  While trying to change the line 141, #define CUSTOM_MACHINE_NAME "Ender3V2-SKRME3V3-MM", in the Marlin > Configuration.h file so that firmware displays my " Ender 5 Pro " name I ran into the compiling issue noted below.  I have listed the steps I used and their results.

Source code compiling issues.

1.  	I Downloaded the current source code zip file.  Ender3V2S1-20220716.zip
	to my "C: Current Source Code" folder.

2.	I Unzip to folder Ender3V2S1-20220716 in my "C: Current Source Code"
	folder.

3.	I Started " Visual Studio Code " & open folder Ender3V2S1-20220716

4.	After loading is complete, and "prior to any editing" I selected the "PlatformIO: Build" check mark and compiled.

5.	The Result:  	Environment      	Status    	Duration
		STM32G0B1RE_btt  	FAILED    	00:00:01.461

The Compile Failed with the "Terminal Pane" showing this error message in red:

buildroot/share/PlatformIO/scripts/../../../../Marlin/src/inc/../lcd/e3v2/proui/
dwin_defines.h:113:10: fatal error: ../../../core/types.h:   No such file or directory

I Followed this path list:  Marlin > src > lcd > e3v2 > proui > dwin_defines.h
and found the following lines are underlined in Red Squiggles

112	#include <stddef.h>
113	#include "../../../core/types.h"
114	#include "../common/dwin_color.h"

Is there something I need to correct in my compiling efforts?
Thank you for your thoughts and inputs.

Regards
I wish 2

**Comment by mriscoc** [comment]
You can find more configuration files to match your hardware here: https://github.com/mriscoc/Special_Configurations

**Reply by I-wish-2** [reply]
Thank you for your reply.  Sorry I was not so clear.  I will try again.
If I open the current Ender3V2S1-20220716 source code in the Visual Studio Code application and did not edit anything.  Should it compile with no errors?
Thank You again
Doug

**Reply by mriscoc** [reply]
Yes, it must be compiled, but the configuration files will not match your current configuration.

Use the updated sources from https://github.com/mriscoc/Ender3V2S1 and can copy the configuration files that matches your hardware from the special configurations' repository

**Reply by I-wish-2** [reply]
Thank you again Miguel,
I'm realizing that this effort is just way over my head.  I thought it was as simple as the first edit I did and compiled with the 20220527 release.  I was able to edit the custom machine name and compile it and then load the bin file to the Ender3 V2 and the name that was displayed was changed.

Not so with the current source code zip file and I'm sorry to be bothering you as I see that this level of editing is now over my head.  Thank you again for trying to help.
My Best Regards
Doug
